mirascope/beta/openai/realtime/realtime.py

```diff
@@ -26,41 +27,75 @@
     async_audio_to_item_create_event,
     audio_chunk_to_audio_segment,
 )
+from mirascope.core import BaseTool
+from mirascope.core.base._utils import (
+    convert_base_model_to_base_tool,
+    convert_function_to_base_tool,
+)
 
 from ._utils._protocols import ReceiverFunc, SenderFunc
+from .tool import FunctionCallArguments, OpenAIRealtimeTool, RealtimeToolParam
 
-ResponseType: TypeAlias = Literal[
-    "text",
-    "audio",
+ChunkResponseType: TypeAlias = Literal[
     "text_chunk",
     "audio_chunk",
-    "audio_transcript",
     "audio_transcript_chunk",
+    "tool_chunk",
 ]
+TextResponseType: TypeAlias = Literal[
+    "text",
+    "audio_transcript",
+]
+AudioResponseType: TypeAlias = Literal["audio"]
+ToolResponseType: TypeAlias = Literal["tool"]
+ResponseType: TypeAlias = (
+    ChunkResponseType | TextResponseType | AudioResponseType | ToolResponseType
+)
 
 
 _ResponseT = TypeVar("_ResponseT")
 _ConnectionT = TypeVar("_ConnectionT")
+_BaseToolT = TypeVar("_BaseToolT", bound=BaseTool)
+_ToolSchemaT = TypeVar("_ToolSchemaT")
+BaseToolT = TypeVar("BaseToolT", bound=BaseTool)
 # TODO: Improve the type of response
 Response: TypeAlias = Any
+CallId: TypeAlias = str
+ResponseId: TypeAlias = str
 
 
 class RawMessage(TypedDict, total=False):
     event_id: str
     type: str
     item: NotRequired[dict]
     delta: NotRequired[str]
+    response_id: NotRequired[str]
+    call_id: NotRequired[str]
+    response: NotRequired[dict]
 
 
 class Sender(TypedDict):
-    func: SenderFunc[_ResponseT]
+    func: SenderFunc[_ResponseT, OpenAIRealtimeTool]
     wait_for_text_response: bool
     wait_for_audio_transcript_response: bool
+    tools: list[type[OpenAIRealtimeTool] | Callable]
 
 
 NotSet = object()
```

Comment:
i don't really get why we need this? can't the user just not set it? what is this doing?

Reply:
This is where you save what is passed as a sender decorator option.
Should the tool be set per top level `session.update` rather than per sender (`response.create`)?
```
    def sender(
        self,
        wait_for_text_response: bool = False,
        wait_for_audio_transcript_response: bool = False,
        tools: list[type[OpenAIRealtimeTool] | Callable] | None = None,
```